src/middleware.ts

```diff
@@ -248,7 +283,15 @@ export const persist = <S extends State>(
       blacklist.forEach((key) => delete state[key])
     }
 
-    return storage?.setItem(name, await serialize({ state, version }))
+    return thenableSerialize({ state, version }).then((serializedValue) => {
+      if (storage) {
+        storage.setItem(name, serializedValue)
+      } else {
+        console.warn(
+          `Persist middleware: unable to update ${name}, the given storage is currently unavailable.`
+        )
+      }
+    })
```

Comment:
```suggestion
    }).catch((e) => { throw e })
```
I'm pretty sure we want this, no?

Reply:
I am not sure if it's working in a `sync` manner. let me check that

Reply:
I think this magically works.

Reply:
👍